### DISCUSSION TITLE
why does `--type toml` includes results with `.lock` ?

### DISCUSSION THREAD

**Discussion by mtb0x1** [BurntSushi/ripgrep#2537] [Q&A] [answered]
Hi,

I am trying to find all `.toml` files containing a pattern  :   

`rg -l -e "dep*" --type toml .`

Results : 
 >Cargo.toml 
 >Cargo.lock

if this is the expected behavior, how and where `type` and `extension` equivalence is set ?

Thanks

**Comment by BurntSushi** [comment, marked as answer]
It's because `Cargo.lock` uses a TOML format. The file type filter is meant to be a model of what the underlying file actually is, and not necessarily an abbreviation for a specific extension. For example, the `c` type includes both `*.c` and `*.h`. They are both C files.

This is why the `toml` type uses `Cargo.lock` spelled out specifically and not `*.lock`, since not all `lock` files use the TOML format.

See:

https://github.com/BurntSushi/ripgrep/blob/a7f1276021df2217dead1481b2c2b38595ed8fb3/crates/ignore/src/default_types.rs#L262

**Reply by mtb0x1** [reply]
thanks @BurntSushi, do I need to add `--type-not lock` in my case or there is a simpler solution like `--extension` ?

**Reply by BurntSushi** [reply]
Sorry, what question are you asking? It seemed like you were asking *why* `Cargo.lock` was included in the `toml` type. But I gather from this comment that you are now asking how to use the `toml` type while simultaneously ignoring `Cargo.lock`. Is that right?

If so, just add a glob rule excluding it. For example, `-ttoml -g '!*.lock'`. Or, you can use the existing `lock` type, i.e., `-ttoml -Tlock`.

**Reply by mtb0x1** [reply]
sorry for the confusion, thanks !
